From 3231f23c31896b1da601e7a0704443d1181a0604 Mon Sep 17 00:00:00 2001
From: Pino Toscano <pino@kde.org>
Date: Mon, 25 Nov 2013 23:26:57 +0100
Subject: [PATCH] tests: split init/tearup

use initTestCase and cleanupTestCase for setting up the tests;
these will be useful later
---
 Tests/testcsvparser.cpp | 6 ++++++
 Tests/testcsvparser.h   | 2 ++
 Tests/testfwparser.cpp  | 6 ++++++
 Tests/testfwparser.h    | 2 ++
 4 files changed, 16 insertions(+)

diff --git a/Tests/testcsvparser.cpp b/Tests/testcsvparser.cpp
index e6b7d23..3547326 100644
--- a/Tests/testcsvparser.cpp
+++ b/Tests/testcsvparser.cpp
@@ -26,6 +26,9 @@
 #include "testcsvparser.h"
 
 TestCSVParser::TestCSVParser(): QObject() {
+}
+
+void TestCSVParser::initTestCase() {
   /*
    * Justification for doing this instead of simply creating a file:
    * To add/change tests, we'll need to modify 2 places. The file and this class.
@@ -113,6 +116,9 @@ TestCSVParser::TestCSVParser(): QObject() {
 
 TestCSVParser::~TestCSVParser()
 {
+}
+
+void TestCSVParser::cleanupTestCase() {
   delete test_parser_;
 }
 
diff --git a/Tests/testcsvparser.h b/Tests/testcsvparser.h
index d8b594f..7cb33cf 100644
--- a/Tests/testcsvparser.h
+++ b/Tests/testcsvparser.h
@@ -30,6 +30,8 @@ class TestCSVParser: public QObject {
   TestCSVParser();
   ~TestCSVParser();
  private slots:
+  void initTestCase();
+  void cleanupTestCase();
   void CSVMixedInputs();
   void CSVQuotesInQuotes();
   void CSVEmptyRow();
diff --git a/Tests/testfwparser.cpp b/Tests/testfwparser.cpp
index a58dd6a..78efcb2 100644
--- a/Tests/testfwparser.cpp
+++ b/Tests/testfwparser.cpp
@@ -18,6 +18,9 @@
 #include "testfwparser.h"
 
 TestFWParser::TestFWParser(): QObject() {
+}
+
+void TestFWParser::initTestCase() {
   test_cases_.append(
     "this is an exam ple of 256 cases being tested -3.14       times\n");
   test_cases_.append(
@@ -71,6 +74,9 @@ TestFWParser::TestFWParser(): QObject() {
 
 TestFWParser::~TestFWParser()
 {
+}
+
+void TestFWParser::cleanupTestCase() {
   delete test_parser_;
 }
 
diff --git a/Tests/testfwparser.h b/Tests/testfwparser.h
index 0a86697..280794b 100644
--- a/Tests/testfwparser.h
+++ b/Tests/testfwparser.h
@@ -29,6 +29,8 @@ class TestFWParser: public QObject {
   TestFWParser();
   ~TestFWParser();
  private slots:
+   void initTestCase();
+   void cleanupTestCase();
    void MixedInputs();
    void OnlySpaceRow();
    void NoRow();
-- 
1.8.4.2

